server/test/unit/dbTest/beginTransactionTest.js

```diff
@@ -0,0 +1,105 @@
+// Third-party dependencies
+const { expect, use } = require('chai')
+const { describe, it } = require('mocha')
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+const rewire = require('rewire')
+
+// In-house dependencies
+const { helpers } = require('brave-alert-lib')
+
+const db = rewire('../../../db/db')
+
+// Configure Chai
+use(sinonChai)
+
+const sandbox = sinon.createSandbox()
+
+describe('db.js unit tests: beginTransaction', () => {
+  /* eslint-disable no-underscore-dangle */
+  let poolConnectStub
+  let pgClient
+
+  beforeEach(() => {
+    sandbox.spy(helpers, 'logError')
+    sandbox.spy(helpers, 'log')
+    poolConnectStub = sandbox.stub()
+    db.__set__('pool', { connect: poolConnectStub })
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  describe('when beginTransaction fails the first time because of a deadlock, then successfully connects after retry', () => {
+    let clientStub
+    beforeEach(async () => {
+      clientStub = { query: sandbox.stub() }
+      sandbox.spy(db, 'beginTransaction')
+      poolConnectStub.onCall(0).rejects(new Error('deadlock detected')).onCall(1).resolves(clientStub)
+      pgClient = await db.beginTransaction()
+    })
+
+    it('should log the deadlock error', () => {
+      expect(helpers.logError).to.be.calledWith(`Error running the beginTransaction query: Error: deadlock detected`)
+    })
+
+    it('should log the retry', () => {
+      expect(helpers.log).to.be.calledWith(`Retrying beginTransaction.`)
+    })
+
+    it('should not return null', () => {
+      expect(pgClient).to.not.be.null
+    })
+  })
+
+  describe('when beginTransaction is successful the first time it is called', () => {
+    let clientStub
+    beforeEach(async () => {
+      clientStub = { query: sinon.stub() }
+      sandbox.spy(db, 'beginTransaction')
+      poolConnectStub.onCall(0).resolves(clientStub)
+      pgClient = await db.beginTransaction()
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.logError).to.not.be.called
+    })
+
```

Comment:
I suggest adding a check that the retry is not logged.

Reply:
Added!